wordpress-mobile/WordPress-Android
# Issue/8509 site creation analytics

Fixes #8509 

**Not Ready for review:** This PR should target `feature/master-site-creation`. Wait until #8864 is merged.

Adds analytics events to the new SiteCreation flow.

List of all events with corresponding attributes - p9wMUP-eo-p2

To test:
- walk through the Site Creation flow and make sure all events are being logged

Update release notes:
- [ x ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.
